### DISCUSSION TITLE
Trying to understand the test system

### DISCUSSION THREAD

**Discussion by patins1** [deepjavalibrary/djl#2945] [Development]
Doing a "gradlew build" execute/tests MnistTest.java (and FashionMnistTest.java) but not  TrainMnistTest.java. so I was wondering why not, and how to invoke  TrainMnistTest.java by a gradle command? Also, how to simulate a nightly build (e.g. turning that flag on in the gradle command)?
Thanks

**Comment by frankfliu** [comment]
1. You can use the following command to select a test for a specific module:

```
./gradlew :examples:test --tests ai.djl.examples.training.TrainMnistTest
```
2.  Pass a system property to the gradle:

```
./gradlew :examples:test --tests ai.djl.examples.training.TrainMnistTest -Dnightly=true
```

3. You can run `TrainMnist` as application:

```
./gradlew run -Dmain=ai.djl.examples.training.TrainMnist
```
see: https://docs.djl.ai/master/examples/docs/train_mnist_mlp.html#build-the-project-and-run

**Reply by patins1** [reply]
Thanks!
